src/inmanta/data/__init__.py

```diff
@@ -1329,7 +1331,7 @@ async def persist_on_halt(cls, env: uuid.UUID, connection: Optional[asyncpg.conn
             Persists paused state when halting all agents.
         """
         await cls._execute_query(
-            f"UPDATE {cls.table_name()} SET unpause_on_resume=NOT paused WHERE environment=$1",
+            f"UPDATE {cls.table_name()} SET unpause_on_resume=NOT paused WHERE environment=$1 AND unpause_on_resume IS NULL",
```

Comment:
@arnaudsjs I added an additional constraint to this query as well to prevent corruption at the lowest possible level.